man/mysql_config.1
'\" t
.\"     Title: \fBmysql_config\fR
.\"    Author: [FIXME: author] [see http://docbook.sf.net/el/author]
.\" Generator: DocBook XSL Stylesheets v1.78.1 <http://docbook.sf.net/>
.\"      Date: 09/18/2015
.\"    Manual: MySQL Database System
.\"    Source: MySQL 5.6
.\"  Language: English
.\"
.TH "\FBMYSQL_CONFIG\FR" "1" "09/18/2015" "MySQL 5\&.6" "MySQL Database System"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.\" mysql_config
.SH "NAME"
mysql_config \- display options for compiling clients
.SH "SYNOPSIS"
.HP \w'\fBmysql_config\ \fR\fB\fIoptions\fR\fR\ 'u
\fBmysql_config \fR\fB\fIoptions\fR\fR
.SH "DESCRIPTION"
.PP
\fBmysql_config\fR
provides you with useful information for compiling your MySQL client and connecting it to MySQL\&. It is a shell script, so it is available only on Unix and Unix\-like systems\&.
.PP
\fBmysql_config\fR
supports the following options\&.
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: cflags option
.\" cflags option: mysql_config
\fB\-\-cflags\fR
.sp
C Compiler flags to find include files and critical compiler flags and defines used when compiling the
libmysqlclient
library\&. The options returned are tied to the specific compiler that was used when the library was created and might clash with the settings for your own compiler\&. Use
\fB\-\-include\fR
for more portable options that contain only include paths\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: cxxflags option
.\" cxxflags option: mysql_config
\fB\-\-cxxflags\fR
.sp
Like
\fB\-\-cflags\fR, but for C++ compiler flags\&. This option was added in MySQL 5\&.6\&.4\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: include option
.\" include option: mysql_config
\fB\-\-include\fR
.sp
Compiler options to find MySQL include files\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: libmysqld-libs option
.\" libmysqld-libs option: mysql_config
.\" mysql_config: embedded option
.\" embedded option: mysql_config
\fB\-\-libmysqld\-libs\fR,
\fB\-\-embedded\fR
.sp
Libraries and options required to link with the MySQL embedded server\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: libs option
.\" libs option: mysql_config
\fB\-\-libs\fR
.sp
Libraries and options required to link with the MySQL client library\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: libs_r option
.\" libs_r option: mysql_config
\fB\-\-libs_r\fR
.sp
Libraries and options required to link with the thread\-safe MySQL client library\&. In MySQL 5\&.6, all client libraries are thread\-safe, so this option need not be used\&. The
\fB\-\-libs\fR
option can be used in all cases\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: plugindir option
.\" plugindir option: mysql_config
\fB\-\-plugindir\fR
.sp
The default plugin directory path name, defined when configuring MySQL\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: port option
.\" port option: mysql_config
.\" TCP/IP
\fB\-\-port\fR
.sp
The default TCP/IP port number, defined when configuring MySQL\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: socket option
.\" socket option: mysql_config
\fB\-\-socket\fR
.sp
The default Unix socket file, defined when configuring MySQL\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: variable option
.\" variable option: mysql_config
\fB\-\-variable=\fR\fB\fIvar_name\fR\fR
.sp
Display the value of the named configuration variable\&. Permitted
\fIvar_name\fR
values are
pkgincludedir
(the header file directory),
pkglibdir
(the library directory), and
plugindir
(the plugin directory)\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysql_config: version option
.\" version option: mysql_config
\fB\-\-version\fR
.sp
Version number for the MySQL distribution\&.
.RE
.PP
If you invoke
\fBmysql_config\fR
with no options, it displays a list of all options that it supports, and their values:
.sp
.if n \{\
.RS 4
.\}
.nf
shell> \fBmysql_config\fR
Usage: /usr/local/mysql/bin/mysql_config [options]
Options:
  \-\-cflags         [\-I/usr/local/mysql/include/mysql \-mcpu=pentiumpro]
  \-\-cxxflags       [\-I/usr/local/mysql/include/mysql \-mcpu=pentiumpro]
  \-\-include        [\-I/usr/local/mysql/include/mysql]
  \-\-libs           [\-L/usr/local/mysql/lib/mysql \-lmysqlclient
                    \-lpthread \-lm \-lrt \-lssl \-lcrypto \-ldl]
  \-\-libs_r         [\-L/usr/local/mysql/lib/mysql \-lmysqlclient_r
                    \-lpthread \-lm \-lrt \-lssl \-lcrypto \-ldl]
  \-\-plugindir      [/usr/local/mysql/lib/plugin]
  \-\-socket         [/tmp/mysql\&.sock]
  \-\-port           [3306]
  \-\-version        [5\&.6\&.11]
  \-\-libmysqld\-libs [\-L/usr/local/mysql/lib/mysql \-lmysqld
                    \-lpthread \-lm \-lrt \-lssl \-lcrypto \-ldl \-lcrypt]
  \-\-variable=VAR   VAR is one of:
          pkgincludedir [/usr/local/mysql/include]
          pkglibdir     [/usr/local/mysql/lib]
          plugindir     [/usr/local/mysql/lib/plugin]
.fi
.if n \{\
.RE
.\}
.PP
You can use
\fBmysql_config\fR
within a command line using backticks to include the output that it produces for particular options\&. For example, to compile and link a MySQL client program, use
\fBmysql_config\fR
as follows:
.sp
.if n \{\
.RS 4
.\}
.nf
gcc \-c `mysql_config \-\-cflags` progname\&.c
gcc \-o progname progname\&.o `mysql_config \-\-libs`
.fi
.if n \{\
.RE
.\}
.SH "COPYRIGHT"
.br
.PP
Copyright \(co 1997, 2015, Oracle and/or its affiliates. All rights reserved.
.PP
This documentation is free software; you can redistribute it and/or modify it only under the terms of the GNU General Public License as published by the Free Software Foundation; version 2 of the License.
.PP
This documentation is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
.PP
You should have received a copy of the GNU General Public License along with the program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA or see http://www.gnu.org/licenses/.
.sp
.SH "SEE ALSO"
For more information, please refer to the MySQL Reference Manual,
which may already be installed locally and which is also available
online at http://dev.mysql.com/doc/.
.SH AUTHOR
Oracle Corporation (http://dev.mysql.com/).


Docs/ChangeLog
commit 0f6cfdf6a16c7c5806786a194cd4f40a2eb3a9e1
Author: Robert Golebiowski <robert.golebiowski@oracle.com>
Date:   Fri Sep 18 11:18:25 2015 +0200

    Updated yassl to yassl-2.3.8
    
    (cherry picked from commit 7f9941eab55ed672bfcccd382dafbdbcfdc75aaa)

commit 1df863e9e8e43e3a96fd98a28cc3194449cceb94
Author: Robert Golebiowski <robert.golebiowski@oracle.com>
Date:   Fri Sep 18 10:35:09 2015 +0200

    Bug #21025377 CAN'T CONNECT TO SSL ENABLED SERVER FIRST 30 SEC AFTER
    INITIAL STARTUP
    
    Updated yassl to yassl-2.3.7e
    
    (cherry picked from commit 6e21c8c04b922bdb60b6a7c174709d2e1bdd3618)

commit 7516bc111acac1292ca0fbfd97ff07970fba064e
Author: Venkatesh Duggirala <venkatesh.duggirala@oracle.com>
Date:   Fri Sep 11 18:56:07 2015 +0530

    Bug#21816399	5.6.24: LOST DATA DURING MASTER RESTART IF PARTIAL TRANSACT HAS BEEN DOWNLOADED
    
    Problem: If dump thread is killed (during shutdown/explicit kill) during
    dumping a non-active binary log, events in that binary log are getting
    skipped completely and not getting replicated to slave.
    
    Analysis: During bug#19975697 fix, we have identified a case where a dump
    thread is unnecessarily looping through a binary log file even if it is
    in killed state. Problem is big enough if the binary log file is huge
    Hence we fixed it by adding thd->killed flag check in while condition.
    But after the while loop is exited, it was not checked whether while loop
    was broken due to thd->killed condition or not. Earlier the code after
    the while loop assumes that the loop processes full binary log file and
    dump thread can be safely moved to next binary log.
    But with this new condition, it is not the case.
    
    Fix: A condition "if (thd->killed)" after the while loop and should exit
    the dump thread without doing any work.
    
    (cherry picked from commit 97d9f26d153e1aa688a5328c5e691953a2761e4d)

commit fbdcb9954d0e36be51a6b559f991e3601f926695
Author: Lars Tangvald <lars.tangvald@oracle.com>
Date:   Fri Sep 4 15:15:10 2015 +0200

    Created default.release for firewall plugin so tests will be run for release build
    
    (cherry picked from commit 88db0500a0e913d65164422b01c9da67350eb4f3)

commit 7f8acabfca414a3e03a727f7fc7be2703168425e
Author: Balasubramanian Kandasamy <balasubramanian.kandasamy@oracle.com>
Date:   Fri Sep 4 17:12:02 2015 +0530

            Bug#21775221 ADD NUMACTL-DEVEL AS BUILDREQ TO RPM SPEC FILES
    
            Description:
    
            Set Numa Mempolicy for optimum mysqld performance added
            libnuma.so as build prerequisite on Linux.
    
            To build with support for this feature numactl devel files are required.
    
            Fix:
    
             Added numactl-devel as build pre-requisite for OL/RHEL/Fedora/docker
             and libnuma-devel for SLES in the spec file.
    
    (cherry picked from commit 18fbfc582f9490d6632c0152d51a3ad19f9660a0)

commit a5c8eb8cb78983413b29c4e05f96597151c1994a
Author: Tor Didriksen <tor.didriksen@oracle.com>
Date:   Fri Sep 4 13:36:18 2015 +0200

    Bug#21785074 BUILD BREAKS IF COMPILED AGAINST OLDER NUMACTL HEADERS
    
    Fix: extend the cmake check with new numa features,
    do not set HAVE_LIBNUMA if built against older numa headers.

commit 40b33206d217239a5f4e4113d487010c851554ef
Author: Venkatesh Duggirala <venkatesh.duggirala@oracle.com>
Date:   Fri Sep 4 11:26:53 2015 +0530

    Bug #19286708	REPLICATION BROKEN AFTER CREATION OF SCHEDULED EVENTS
    
    Reverting patch due to a regression

commit 4facaa3572c0f907d922c53730c08ad68cc60054
Author: Balasubramanian Kandasamy <balasubramanian.kandasamy@oracle.com>
Date:   Thu Sep 3 17:52:13 2015 +0530

        Bug#21775221 ADD NUMACTL-DEVEL AS BUILDREQ TO RPM SPEC FILES
    
        Description:
    
        Set Numa Mempolicy for optimum mysqld performance added
        libnuma.so as build prerequisite on Linux.
    
        To build with support for this feature numactl devel files are required.
    
        Fix:
    
         Added numactl-devel as build pre-requisite for OL/RHEL/Fedora/docker
         and libnuma-devel for SLES in the spec file.
    
    (cherry picked from commit 45d038b07faf7fd2d508fff77ed9322c3a3c3f0c)

commit bb6b072e6a7689c2b7c642ba1f35655090b6b551
Author: Tor Didriksen <tor.didriksen@oracle.com>
Date:   Thu Sep 3 10:16:24 2015 +0200

    Bug#21774859 CMAKE CHECKS FOR NUMA SUPPORT TOO WEAK
    
    The check for numa availability implemented by the patch for
    Bug #18871046 SET NUMA MEMPOLICY FOR OPTIMUM MYSQLD PERFORMANCE
    was wrong: it forgot to restore CMAKE_REQUIRED_LIBRARIES after the check.
    
    This affects all subsequent platform feature tests that rely on cmakes
    CHECK_CXX_SOURCE_COMPILES. In particular: on platforms which do not have
    numa support, the test for 'HAVE_PEERCRED' will also fail, and the
    'auth_socket' plugin will not be built.

commit 56c2c3d4ecdab0b2ab1bab8abd01c5de509e6550
Merge: 928440d 282ed17
Author: Murthy Narkedimilli <murthy.narkedimilli@oracle.com>
Date:   Mon Aug 31 20:43:30 2015 +0200

    Empty version change upmerge

commit 282ed178bb3e456cbdbc6b98032c36ccaa867f8f
Author: Murthy Narkedimilli <murthy.narkedimilli@oracle.com>
Date:   Mon Aug 31 20:36:03 2015 +0200

    Raise version number after cloning 5.5.46

commit 928440d4bb1c0f436dd4a23bc95fd7d416a3350b
Author: Ajo Robert <ajo.robert@oracle.com>
Date:   Mon Aug 31 15:08:59 2015 +0530

    Revert "Bug #20201006 spamming show processlist prevents old"
    
    This reverts commit 883e25e945f304f05130ee541f7962004ed9aa90.

commit 0eb0cf1dbe3b10724b4fdfe80298fc51f9a4c8c9
Author: Jon Olav Hauglid <jon.hauglid@oracle.com>
Date:   Tue Jun 30 15:17:57 2015 +0200

    Bug#21682439: BACKPORT OF BUG#20928289 TO MYSQL 5.6
    
    Backport from mysql-5.7 to mysql-5.6 of:
    
    Bug#20928289: DEFAULT CMAKE EXCLUDES UTF8_5624_1 CHARSET,
                  REQUIRED FOR MAIN.CTYPE_LDML
    

scripts/mysql_config.sh
#!/bin/sh
# Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA

# This script reports various configuration settings that may be needed
# when using the MySQL client library.

which ()
{
  IFS="${IFS=   }"; save_ifs="$IFS"; IFS=':'
  for file
  do
    for dir in $PATH
    do
      if test -f $dir/$file
      then
        echo "$dir/$file"
        continue 2
      fi
    done
    echo "which: no $file in ($PATH)"
    exit 1
  done
  IFS="$save_ifs"
}

#
# If we can find the given directory relatively to where mysql_config is
# we should use this instead of the incompiled one.
# This is to ensure that this script also works with the binary MySQL
# version

fix_path ()
{
  var=$1
  shift
  for filename
  do
    path=$basedir/$filename
    if [ -d "$path" ] ;
    then
      eval "$var"=$path
      return
    fi
  done
}

get_full_path ()
{
  file=$1

  # if the file is a symlink, try to resolve it
  if [ -h $file ];
  then
    file=`ls -l $file | awk '{ print $NF }'`
  fi

  case $file in
    /*) echo "$file";;
    */*) tmp=`pwd`/$file; echo $tmp | sed -e 's;/\./;/;' ;;
    *) which $file ;;
  esac
}

me=`get_full_path $0`

# Script might have been renamed but assume mysql_<something>config<something>
basedir=`echo $me | sed -e 's;/bin/mysql_.*config.*;;'`

ldata='@localstatedir@'
execdir='@libexecdir@'
bindir='@bindir@'

# If installed, search for the compiled in directory first (might be "lib64")
pkglibdir='@pkglibdir@'
pkglibdir_rel=`echo $pkglibdir | sed -e "s;^$basedir/;;"`
fix_path pkglibdir $pkglibdir_rel @libsubdir@/mysql @libsubdir@

plugindir='@pkgplugindir@'
plugindir_rel=`echo $plugindir | sed -e "s;^$basedir/;;"`
fix_path plugindir $plugindir_rel @libsubdir@/mysql/plugin @libsubdir@/plugin

pkgincludedir='@pkgincludedir@'
if [ -f "$basedir/include/mysql/mysql.h" ]; then
  pkgincludedir="$basedir/include/mysql"
elif [ -f "$basedir/include/mysql.h" ]; then
  pkgincludedir="$basedir/include"
fi

version='@VERSION@'
socket='@MYSQL_UNIX_ADDR@'
ldflags='@LDFLAGS@'

if [ @MYSQL_TCP_PORT_DEFAULT@ -eq 0 ]; then
  port=0
else
  port=@MYSQL_TCP_PORT@
fi

# Create options 
# We intentionally add a space to the beginning and end of lib strings, simplifies replace later
libs=" $ldflags -L$pkglibdir @RPATH_OPTION@ -lwebscalesqlclient @ZLIB_DEPS@ @NON_THREADED_LIBS@"
libs="$libs @openssl_libs@ @STATIC_NSS_FLAGS@ "
libs="$libs @QUOTED_CMAKE_C_LINK_FLAGS@"
libs_r=" $ldflags -L$pkglibdir @RPATH_OPTION@ -lfbmysqlclient @ZLIB_DEPS@ @CLIENT_LIBS@ @openssl_libs@ "
libs_r="$libs_r @QUOTED_CMAKE_C_LINK_FLAGS@"
embedded_libs=" $ldflags -L$pkglibdir @RPATH_OPTION@ -lmysqld @LIBDL@ @ZLIB_DEPS@ @LIBS@ @WRAPLIBS@ @openssl_libs@ "
embedded_libs="$embedded_libs @QUOTED_CMAKE_CXX_LINK_FLAGS@"

cflags="-I$pkgincludedir @CFLAGS@ " #note: end space!
cxxflags="-I$pkgincludedir @CXXFLAGS@ " #note: end space!
include="-I$pkgincludedir"

# Remove some options that a client doesn't have to care about
for remove in DDBUG_OFF DSAFE_MUTEX DFORCE_INIT_OF_VARS \
              DEXTRA_DEBUG DHAVE_purify O 'O[0-9]' 'xO[0-9]' 'W[-A-Za-z]*' \
              'mtune=[-A-Za-z0-9]*' 'mcpu=[-A-Za-z0-9]*' 'march=[-A-Za-z0-9]*' \
              unroll2 ip mp restrict 'specs=[A-Za-z0-9/]*\.spec'
do
  # The first option we might strip will always have a space before it because
  # we set -I$pkgincludedir as the first option
  cflags=`echo "$cflags"|sed -e "s/ -$remove  */ /g"` 
  cxxflags=`echo "$cxxflags"|sed -e "s/ -$remove  */ /g"` 
done
cflags=`echo "$cflags"|sed -e 's/ *\$//'` 
cxxflags=`echo "$cxxflags"|sed -e 's/ *\$//'` 

# Same for --libs(_r)
for remove in lmtmalloc static-libcxa i-static static-intel
do
  # We know the strings starts with a space
  libs=`echo "$libs"|sed -e "s/ -$remove  */ /g"` 
  libs_r=`echo "$libs_r"|sed -e "s/ -$remove  */ /g"` 
  embedded_libs=`echo "$embedded_libs"|sed -e "s/ -$remove  */ /g"` 
done

# Strip trailing and ending space if any, and '+' (FIXME why?)
libs=`echo "$libs" | sed -e 's;  \+; ;g' | sed -e 's;^ *;;' | sed -e 's; *\$;;'`
libs_r=`echo "$libs_r" | sed -e 's;  \+; ;g' | sed -e 's;^ *;;' | sed -e 's; *\$;;'`
embedded_libs=`echo "$embedded_libs" | sed -e 's;  \+; ;g' | sed -e 's;^ *;;' | sed -e 's; *\$;;'`

usage () {
        cat <<EOF
Usage: $0 [OPTIONS]
Options:
        --cflags         [$cflags]
        --cxxflags       [$cxxflags]
        --include        [$include]
        --libs           [$libs]
        --libs_r         [$libs_r]
        --plugindir      [$plugindir]
        --socket         [$socket]
        --port           [$port]
        --version        [$version]
        --libmysqld-libs [$embedded_libs]
        --variable=VAR   VAR is one of:
                pkgincludedir [$pkgincludedir]
                pkglibdir     [$pkglibdir]
                plugindir     [$plugindir]
EOF
        exit 1
}

if test $# -le 0; then usage; fi

while test $# -gt 0; do
        case $1 in
        --cflags)  echo "$cflags" ;;
        --cxxflags)echo "$cxxflags";;
        --include) echo "$include" ;;
        --libs)    echo "$libs" ;;
        --libs_r)  echo "$libs_r" ;;
        --plugindir) echo "$plugindir" ;;
        --socket)  echo "$socket" ;;
        --port)    echo "$port" ;;
        --version) echo "$version" ;;
        --embedded-libs | --embedded | --libmysqld-libs) echo "$embedded_libs" ;;
        --variable=*)
          var=`echo "$1" | sed 's,^[^=]*=,,'`
          case "$var" in
            pkgincludedir) echo "$pkgincludedir" ;;
            pkglibdir) echo "$pkglibdir" ;;
            plugindir) echo "$plugindir" ;;
            *) usage ;;
          esac
          ;;
        *)         usage ;;
        esac

        shift
done

#echo "ldata: '"$ldata"'"
#echo "execdir: '"$execdir"'"
#echo "bindir: '"$bindir"'"
#echo "pkglibdir: '"$pkglibdir"'"
#echo "pkgincludedir: '"$pkgincludedir"'"
#echo "version: '"$version"'"
#echo "socket: '"$socket"'"
#echo "port: '"$port"'"
#echo "ldflags: '"$ldflags"'"
#echo "client_libs: '"$client_libs"'"

exit 0
